src/parser.js

```diff
@@ -0,0 +1,59 @@
+require('dotenv').config();
+const bent = require('bent');
+const jsdom = require('jsdom');
+
+const request = bent('string');
+const { JSDOM } = jsdom;
+
+/*
+* getData() returns Promise { <pending> }
+* It gets the data from 'pre' tag in a provided url
+* Splits the data into lines so its easier to process as a string array
+* That data is then returned as a Promise
+*/
+function getData() {
+  const list = request(process.env.FEED_URL)
+    .then((data) => {
+      const dom = new JSDOM(data);
+      let feedList = '';
+      feedList = dom.window.document.querySelector('pre').textContent;
+      return feedList.split(/\r\n|\r|\n/);
+    }).catch((err) => { throw err; });
+
+  return list;
+}
+
+/*
+* parseData() returns Promise { <pending> }
+* It gets the data from getData function
+* Then processes it to remove square brackets from links and 'name=' in front of a name
+* That data is then returned as a Promise
+*/
+function parseData() {
+  const nameCheck = /^name/i;
+  const commentCheck = /^#/;
+
+  let feed = '';
+  let line = '';
+
+  const parsedData = getData().then((data) => {
+    data.forEach((element) => {
+      if (!commentCheck.test(element)) {
+        if (element.startsWith('[')) {
+          line = element.replace(/([\[\]']+)/g, '');
+          feed += `${line}\n`;
+        }
+        if (nameCheck.test(element)) {
+          line = element.replace(/^\s*name\s*=\s*/, '');
+          feed += `${line}\n`;
+        }
+      }
+      console.log(feed);
+      return feed;
```

Comment:
We need to think about this flow a bit.  You're in a `then()` resolving your data, and want to parse it into items that can be returned.  Within this, you're in a `forEach` and you're returning within the callback to the `forEach`, which doesn't really do anything, since nothing will receive what you return.

Your `then()` should probably have a return before it exists, so whatever code consumes this Promise can do something with the processed data.